webui/react/src/routes/index.ts

```diff
@@ -1,8 +1,9 @@
 import { RouteProps } from 'react-router';
 
-import Authentication from 'pages/Authentication';
 import Dashboard from 'pages/Dashboard';
 import Determined from 'pages/Determined';
+import SignIn from 'pages/SignIn';
+import SignOut from 'pages/SignOut';
```

Comment:
it would make sharing logic between the two harder but on the other hand, does put the code in a separate file

Reply:
The logout doesn't actually have a view but it separates it the main function of signing out into its own unit (in form of a route). With the addition of handling the auto sign in, the login page was being to grow unwieldy. By having the logout logic inside login in addition to auto auth detection, because there were a lot of redirects, if the conditions are right, it created a constant cycle of redirects. This could probably be fixed, but it was not easy to navigate. Having this separation significantly simplified the redirects.

Reply:
ah yes, meant to add details on the cookie polling / checking on the sign in page. So I tried that route and realized that this approach created an infinite redirect.

1) On the Sign in page, check to find that `auth` cookie is set, so redirect to an auth page.
2) Within the `Route` component, a `needsAuth` is encountered and `auth.isAuthenticated` is false because cookie has not been verified. Redirects back to login.
3) GOTO LINE 1) good old BASIC

Reply:
talked with @hkang1 about this, it looked like it might still be possible to do this but IMO it would also work to go ahead with the current approach and get this feature and the PR landed sooner and revisit the cookie approach separately.